### ISSUE TITLE
Add own online sources

### ISSUE THREAD

**Issue by george-roussos** [theothernt/AerialViews#116]
Hello and thanks for this wonderful app!

Is there a way the end user can add their own remote sources? I thought it would be nice if we could provide a path to a .JSON file a la [this](https://github.com/theothernt/AerialViews/blob/master/app/src/main/res/raw/comm1.json) and then the app can  stream them. Thanks again!

**Comment by theothernt** [comment]
Yes, this is the plan at some point. 

I've always found text entry on Android TV to be awkward so I've tried to avoid situations where the user has to enter a lot of information. That said, it would be nice to have the ability to add video feeds - and not just for the list of videos, but the fact that the video manifest files also contain descriptions, etc.

Also, I might add local support, meaning that if you a) download all the videos and b) place the manifest (manifest.json or videos.json etc) inside the same folder, Aerial Views could pick that up and show descriptions for those videos.

**Comment by pbvdven** [comment]
Could it also be possible to use a own live stream url so i can use my security camera bridsview stream as screensaver. I use this for streaming my camera feed https://github.com/AlexxIT/go2rtc

**Comment by theothernt** [comment]
> Could it also be possible to use a own live stream url so i can use my security camera bridsview stream as screensaver. I use this for streaming my camera feed https://github.com/AlexxIT/go2rtc

Yes, that should be possible if **go2rtc** if configured to output a video stream or format that ExoPlayer supports eg. MP4 or HLS.

Be aware - if you have 2 videos streams, Aerial Views play the first stream forever as the stream doesn't end. Use the option Settings > Playlist > Limit video length to skip to the next stream after 30 seconds, for example.

Also, just curious, can **go2rtc** combine the streams of multiple cameras into one?

**Comment by pbvdven** [comment]
No i use frigate for that they have a option called birdview it combines all the cameras into one stream. You can show them all at once in squares or in cycle. Okay thanks im going to try setting it up. Can you give any direction where i can put the stream url im only seeing local file or samba as an option. Thanks

**Comment by theothernt** [comment]
> Okay thanks im going to try setting it up. Can you give any direction where i can put the stream url im only seeing local file or samba as an option. Thanks

Apologies - I didn't explain properly. The feature hasn't been made yet, I'm just asking questions to try and figure out how people will expect to use it.

It might be possible for me to make a test version of the feature in the next week or two.

**Comment by pbvdven** [comment]
Aha that explains a lot. a test version would be great if i can be of any help let me know. Thanks

**Comment by guitarbeat** [comment]
I would love this feature, i want to setup a youtube playlist as my screensaver.

**Comment by theothernt** [comment]
As I haven't made any progress on the feature - your best bet @guitarbeat, would be to use a tool like [yt-dlp](https://github.com/yt-dlp/yt-dlp) to download the playlist of videos and play them locally or from a network share.

YT integration will be tricky as it might get the app banned.

**Comment by theothernt** [comment]
Custom feed support has been added in the latest public beta 5 if you anyone would like to try it?

It supports custom entreies.json feeds which follow this format: https://github.com/AerialScreensaver/AerialCommunity/blob/master/entries.json

Also, you can use direct RTSP URLs.

Let me know if you have any questions?

**Comment by theothernt** [comment]
Aerial Views v1.8.0 has been released on GitHub. It should be released on Google Play in a few days.
